GitUI/UserControls/RevisionGridClasses/DvcsGraph.cs

```diff
@@ -1276,8 +1276,8 @@ protected override void OnKeyDown(KeyEventArgs e)
 
         private sealed class Node
         {
-            public readonly List<Junction> Ancestors = new List<Junction>();
-            public readonly List<Junction> Descendants = new List<Junction>();
+            public readonly List<Junction> Ancestors = new List<Junction>(capacity: 2);
+            public readonly List<Junction> Descendants = new List<Junction>(capacity: 2);
```

Comment:
I can understand a node having only two ancestors, but can it not have more than 2 descendants?

Reply:
When loading the GE repo in GE, this change avoids ~2MB of redundant heap allocation.

These values don't represent hard limits as the lists can resize if necessary.

The assumption here is that the vast majority of nodes wouldn't have more than two ancestors or descendants.

`List<T>` has a default capacity of 16 (possibly implementation specific), we save 14 bytes for 2 collections on every node. On GE we have just over 10,000 commits.

So for GE on a 64 bit process allocations are reduced by 14*2*10,000*8 = 1,920,000 bytes. 

Maybe we should go further. If we use default capacities of 1 then we would save 2*10,000*8 = 160,000 additional bytes (8% more), but have to resize the array for merge commits etc.

Reply:
Thank you for the explanation 👍